src/core/thread/mle_router.cpp

```diff
@@ -2007,6 +2039,13 @@ otError MleRouter::UpdateChildAddresses(const Message &aMessage, uint16_t aOffse
     {
         uint8_t len;
 
+#if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
+        if (mMaxChildIpAddresses > 0 && storedCount >= mMaxChildIpAddresses)
+        {
+            // Skip remaining address registration entries.
+            break;
+        }
+#endif
```

Comment:
How about instead of  `break`, we do similar check around `error = aChild.AddIp6Address(address);` and skip it.
This way we still go through the code where we log the skipped addresses and clear the address-cache, etc.

Reply:
Good point! Updated, please help to take a look.